opm/simulators/flow/SimulatorFullyImplicitBlackoilEbos.hpp

```diff
@@ -430,10 +431,10 @@ class SimulatorFullyImplicitBlackoilEbos
         report_.success.solver_time += solverTimer_->secsSinceStart();
 
         if (this->grid().comm().rank() == 0) {
-            // Destructively grab the step convergence reports.  The solver
-            // object and the model object contained therein are about to go
-            // out of scope.
-            this->writeConvergenceOutput(solver->model().getStepReportsDestructively());
+            // Grab the step convergence reports that are new since last we were here.
+            const auto& reps = solver_->model().stepReports();
+            this->writeConvergenceOutput(std::vector<StepReport>{reps.begin() + already_reported_steps_, reps.end()});
```

Comment:
It doesn't hurt, of course, but I don't think you explicitly need to call the `vector<>` constructor here.  You should be able to just pass the `initializer_list<>` directly as an argument to `writeConvergenceOutput()`, e.g.,
```C++
this->writeConvergenceOutput({ reps.begin() + ..., reps.end() });
```

Reply:
I chose to use the explicit constructor for clarity, so I don't get too confused when I read the code again in a year...